### PR TITLE
Always load the builtin plugins type extensions

### PR DESCRIPTION
**Pull request by alcuadrado**
Another take on how we load the builtin plugin type extension. This time making sure that we always load them, so we don't have edge cases of incomplete type, at the cost of a runtime import of an empty js file.

There's an explanation this in `packages/hardhat/src/types/builtin-plugin-type-extensions.ts`.

What motivated this change is that @Wodann was hitting the edge cases of the previous approach in one of his tests.

### CODE REVIEW

File: packages/hardhat/src/types/builtin-plugin-type-extensions.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// This is an internal file that needs to be imported by every top-level type
+// file, so that they load the builtin plugin type extensions.
```
[the remark this answers is not in the document]
Comment (kanej) [reply packages/hardhat/src/types/builtin-plugin-type-extensions.ts:2]: This is a genuine exception. We don't need to expand the comment to cover it.

File: packages/hardhat/src/types/builtin-plugin-type-extensions.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+//
+// The trade-off here is that we end up with one unnecessary runtime import of
+// an empty file (this one) if a `src/types/*` file is imported without the
+// `type` annotation (e.g. the CLI's indirectly importing task-related enums),
+// in exchange for better DX.
```
[the remark this answers is not in the document]
Comment (kanej) [reply packages/hardhat/src/types/builtin-plugin-type-extensions.ts:7]: The cost here does not warrant the test guard.

**Review by kanej — APPROVED** [review]
LGTM!

**Comment by Wodann** [comment]
Would it be possible to add a lint or script that validates in CI whether all files in the `src/types/` folder include `import "./builtin-plugin-type-extensions.js";`?

It seems like something that is easy to miss by a developer.
